src/net/sourceforge/kolmafia/textui/Parser.java

```diff
@@ -4765,21 +4805,50 @@ private Token makeComment( final int commentLength )
 			return newToken;
 		}
 
+		private Token peekLastToken()
+		{
+			if ( this.tokens.isEmpty() )
+			{
+				if ( this.previousLine == null )
+				{
+					return null;
+				}
+
+				return this.previousLine.peekLastToken();
+			}
+
+			final Token lastToken = this.tokens.peekLast();
+
+			if ( lastToken instanceof Comment )
+			{
+				// Temporarily remove it before digging deeper
+				this.tokens.removeLast();
+				final Token previousToken = this.peekLastToken();
```

Comment:
I... think it's possible to achieve with iteration, but not having to mutate this.tokens? No, they have to get mutated.

Reply:
Well, Deque has `Deque.descendingIterator()`, which is pretty handy...

Any way of quickly turning an `Iterator` into an `Iterable`? (to easily fit it in a `for` loop)

Reply:
I'm not aware of one. I'd just iterate over it using hasNext / next. It's not that complicated...

Something like...
```
Line line = this;
while (line != null) {
  Iterator<Token> it = this.tokens.descendingIterator();
  while (it.hasNext()) {
    Token token = it.next();
    if (token instanceof Comment) continue;
    return token;
  }
  line = this.previousLine;
}

return null;
```
seems like it would work here.

Reply:
(Not really used to using Iterator...)

How's it look now?

Reply:
Oh, didn't notice you answered... turns out I got the same idea as you

Reply:
Much better, although I assume you didn't mean to keep an unconditional break at line 4841.

Reply:
wait, whoops...